src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6970,17 +6970,22 @@ and TcConstStringExpr cenv (overallTy: OverallTy) env m tpenv (s: string) litera
         | _ -> false
 
     let g = cenv.g
+  
 
-    if isFormat g overallTy.Commit then
-        if literalType = LiteralArgumentType.StaticField then
-            checkLanguageFeatureAndRecover g.langVersion LanguageFeature.NonInlineLiteralsAsPrintfFormat m
+    match isFormat g overallTy.Commit, literalType with
+    | true, LiteralArgumentType.StaticField ->
+        checkLanguageFeatureAndRecover g.langVersion LanguageFeature.NonInlineLiteralsAsPrintfFormat m
+        TcFormatStringExpr cenv overallTy env m tpenv s literalType
 
+    | true, LiteralArgumentType.Inline ->
         TcFormatStringExpr cenv overallTy env m tpenv s literalType
-    else
-        if literalType = LiteralArgumentType.Inline && not (isObjTy g overallTy.Commit) then
-            AddCxTypeEqualsType env.eContextInfo env.DisplayEnv cenv.css m overallTy.Commit g.string_ty
 
-        mkString g m s, tpenv
+    | false, LiteralArgumentType.StaticField ->
+        Expr.Const (TcFieldInit m (ILFieldInit.String s), m, g.string_ty), tpenv
```

Comment:
I think all `TcFieldInit m (ILFieldInit.String s)` will do is return `Const.String s`, so might as well replace this line with `Expr.Const (Const.String s, m, g.string_ty), tpenv` or `mkString g m s`.